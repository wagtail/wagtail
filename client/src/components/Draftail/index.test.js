import {
  wrapWagtailIcon,
  initEditor,
  registry,
  ModalWorkflowSource,
  Link,
  Document,
  ImageBlock,
  EmbedBlock,
} from './index';

describe('Draftail', () => {
  describe('#initEditor', () => {
    beforeEach(() => {
      document.body.innerHTML = '';
    });

    it('works', () => {
      const field = document.createElement('input');
      field.name = 'test';
      field.value = 'null';
      document.body.appendChild(field);

      initEditor('test', {});

      expect(field.draftailEditor).toBeDefined();
    });

    it('onSave', () => {
      const field = document.createElement('input');
      field.name = 'test';
      field.value = 'null';
      document.body.appendChild(field);

      initEditor('test', {});

      field.draftailEditor.saveState();

      expect(field.value).toBe('null');
    });

    it('options', () => {
      const field = document.createElement('input');
      field.name = 'test';
      field.value = 'null';
      document.body.appendChild(field);

      registry.registerPlugin({
        type: 'IMAGE',
        source: () => {},
        block: () => {},
      });

      initEditor('test', {
        entityTypes: [
          { type: 'IMAGE' },
        ],
        enableHorizontalRule: true,
      });

      expect(field.draftailEditor.props).toMatchSnapshot();
    });
  });

  describe('#wrapWagtailIcon', () => {
    it('works', () => {
      expect(wrapWagtailIcon({ icon: 'media' }).icon).toMatchSnapshot();
    });

    it('no icon', () => {
      const type = {};
      expect(wrapWagtailIcon(type)).toBe(type);
    });

    it('array icon', () => {
      const type = { icon: ['M10 10 H 90 V 90 H 10 Z'] };
      expect(wrapWagtailIcon(type)).toBe(type);
    });
  });

  it('#registry', () => expect(registry).toBeDefined());
  it('#ModalWorkflowSource', () => expect(ModalWorkflowSource).toBeDefined());
  it('#Link', () => expect(Link).toBeDefined());
  it('#Document', () => expect(Document).toBeDefined());
  it('#ImageBlock', () => expect(ImageBlock).toBeDefined());
  it('#EmbedBlock', () => expect(EmbedBlock).toBeDefined());
});
