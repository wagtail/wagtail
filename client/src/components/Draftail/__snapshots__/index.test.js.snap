// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draftail #initEditor options 1`] = `
Object {
  "autoCapitalize": null,
  "autoComplete": null,
  "autoCorrect": null,
  "blockTypes": Array [],
  "decorators": Array [],
  "enableHorizontalRule": Object {
    "description": "Horizontal line",
  },
  "enableLineBreak": Object {
    "description": "Line break",
  },
  "entityTypes": Array [
    Object {
      "block": [Function],
      "source": [Function],
      "type": "IMAGE",
    },
  ],
  "inlineStyles": Array [],
  "maxListNesting": 4,
  "onSave": [Function],
  "placeholder": "Write hereâ€¦",
  "rawContentState": null,
  "showRedoControl": Object {
    "description": "Redo",
  },
  "showUndoControl": Object {
    "description": "Undo",
  },
  "spellCheck": true,
  "stateSaveInterval": 250,
  "stripPastedStyles": false,
  "textAlignment": null,
  "textDirectionality": null,
}
`;

exports[`Draftail #wrapWagtailIcon works 1`] = `
<Icon
  className={null}
  name="media"
  title={null}
/>
`;
